server.go

```diff
@@ -0,0 +1,216 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"reflect"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/go-chi/chi/v5"
+	"github.com/invopop/jsonschema"
+)
+
+type Server struct {
+	serverConfig *ServerConfig
+	router       *Router
+}
+
+type ServerConfig struct {
+	name string
+	port int
+}
+
+type RegistrationFunctionSchema struct {
+	InputSchema  *jsonschema.Schema `json:"input"`
+	OutputSchema *jsonschema.Schema `json:"output"`
+	FnName       string             `json:"functionName"`
+	EndPoint     string             `json:"endpoint"`
+}
+
+// Creates and returns an empty Server, with a basic serverConfig
+func Create(name string) *Server {
+
+	return &Server{
+		serverConfig: &ServerConfig{
+			name: name,
+			port: 3000,
+		},
+		router: &Router{
+			routes: []*Route{},
+		},
+	}
+}
+
+func (s *Server) Router() *Router {
+	return s.router
+}
+
+func (server *Server) Start() {
+	// todo: check if router was nil
+	// todo: check if the routes was empty
+	// todo: any additional checks needed ?
+	// todo: check if the server was already started ?
+
+	// create a new internal go-chi router
+	chi_router := chi.NewRouter()
+
+	// read all the routes from the router
+	routes := server.router.routes
+	for _, route := range routes {
+		newFunc := func(w http.ResponseWriter, r *http.Request) {
+			// Decode the JSON request body into the struct instance
+			structPtr := reflect.New(route.handler.inpType).Interface()
+			decoder := json.NewDecoder(r.Body)
+			decoder.DisallowUnknownFields()
+			err := decoder.Decode(structPtr)
+			if err != nil {
+				http.Error(w, "Invalid JSON request body", http.StatusBadRequest)
+				return
+			}
+
+			decodedValue := reflect.ValueOf(structPtr).Elem()
+			decodedType := decodedValue.Type()
+
+			if decodedType != route.handler.inpType {
```

Comment:
Why is this required?

Reply:
This is to check if the parsed body in the request matches with the expected Input type for the handler.
But I feel the decoder.Decode() logic above would do that, right?
Hence, this is not needed ?

Reply:
Yes. It isn't needed